### PR TITLE
Move core data operations to background

### PR DESCRIPTION
**Pull request by bjrne**
It's rather a working solution than anything beautiful but it works for now.

### CODE REVIEW

File: xikolo-ios/ViewControllers/Courses/CourseContentTableViewController.swift [hunk 1]
```diff
@@ -84,11 +84,11 @@ class CourseContentTableViewController: UITableViewController {
         switch segue.identifier! {
         case "ShowVideo":
             let videoView = segue.destination as! VideoViewController
-            videoView.courseItem = courseItem
+            videoView.courseItem = try! CourseItemHelper.getByID(courseItem!.id)
```
Comment (mathebox) [inline xikolo-ios/ViewControllers/Courses/CourseContentTableViewController.swift]: What about error handling?

File: xikolo-ios/ViewControllers/Courses/CourseContentTableViewController.swift [hunk 2]
```diff
@@ -12,7 +12,7 @@ import DZNEmptyDataSet
 
 class CourseContentTableViewController: UITableViewController {
 
-    var course: Course!
+    var cdCourse: Course!
```
Comment (mathebox) [inline xikolo-ios/ViewControllers/Courses/CourseContentTableViewController.swift]: Why "cdCourse"?

Comment (bjrne) [inline xikolo-ios/ViewControllers/Courses/CourseContentTableViewController.swift]: I was waiting for your input on this. I changed this in the last commit to make clear that this is a core data object from the background, not one that I got out of a list view controller. But afterwards I noticed that we should basically have this all the time

File: xikolo-ios/ViewControllers/Courses/CourseContentTableViewController.swift (hunk 1 above)
Comment (bjrne) [inline xikolo-ios/ViewControllers/Courses/CourseContentTableViewController.swift]: It should not be possible the get an error here, but I was also thinking about an appropriate error handling. Do you have an idea?

**Pull request merged**